### ISSUE TITLE
Helm Not Accessible in Rancher after helm init

### ISSUE THREAD

**Issue by aric49** [rancher/rancher#14943]
**Rancher versions:**
rancher/server or rancher/rancher: 2.0.6
rancher/agent or rancher/rancher-agent: 2.0.6

**Infrastructure Stack versions:**
healthcheck:
ipsec:
network-services:
scheduler:
kubernetes (if applicable): 1.10.5

**Docker version: (`docker version`,`docker info` preferred)** 17.03.2-ce

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
```
NAME="RancherOS"
VERSION=v1.4.0
ID=rancheros
ID_LIKE=
VERSION_ID=v1.4.0
PRETTY_NAME="RancherOS v1.4.0"
HOME_URL="http://rancher.com/rancher-os/"
SUPPORT_URL="https://forums.rancher.com/c/rancher-os"
BUG_REPORT_URL="https://github.com/rancher/os/issues"
BUILD_ID=
```

**Helm Version**:  
```
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
```

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**: AWS

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**: Single Node, HA-Kubernetes Cluster

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**: Kubernetes

**Steps to Reproduce:**
1. Spin up Kubernetes cluster with AWS cloud provider
2. Grab the KubeConfig from the Rancher UI
2. Create tiller service account
3. Instantiate Helm using `helm init --service-account tiller`

**Results:**
Any Helm command (helm ls, helm install, helm upgrade) returns the error:

```
error: error upgrading connection: Upgrade request required
```

I believe it has to do with how Rancher is proxying requests into the Kubernetes cluster.   Exposing tiller on host networking or by exposing a node port seems to be working.   I found this related issue:  https://github.com/rancher/rancher/issues/7303  in which the reporting user indicates that their issue was resolved by exporting a nodeport as the $HELM_HOST variable.  However, I think this should work by default.

Please see this corresponding Gist with more verbose details:   https://gist.github.com/aric49/091cad31e6cfd429c715e14d4fb546ea

Any assistance is greatly appreciated!  Thank you in advance.

**Comment by aric49** [comment]
After debugging, I discovered that this issue seems to be related to the Amazon Application LoadBalancer I configured in front of our Rancher server.  I discovered during this time that commands such as `kubectl exec`, `kubectl logs`, and `kubectl port-forward` all seem to fail with the same error message.   I tried experimenting with using an ELB instead of an ALB and the same issue persists.

Eventually, I was able to work around this issue by bypassing the load balancer all together and issue requests directly to the Rancher server.   It appears as though Amazon load balancers are filtering out some traffic that Kubernetes expects. 

I found in the Rancher 1.6 docs a disclaimer about using an ELB as opposed to an ALB, and directing traffic to port 8080 but I believe this is deprecated since it appears that the `rancher/rancher` docker image does not listen on port 8080:     https://rancher.com/docs/rancher/v1.6/en/installing-rancher/installing-server/basic-ssl-config/#enabling-proxy-protocol

Any futher help with this issue is greatly appreciated.   I am somewhat satisfied with the workaround I discovered, but would ultimately like to be able to use some type of load balancer for kubectl traffic as opposed to just WebUI management.
